main.tf

```diff
@@ -51,7 +51,7 @@ resource "ibm_database" "etcd_db" {
   tags              = var.tags
   adminpassword     = var.admin_pass
   service_endpoints = var.service_endpoints
-  configuration     = var.configuration != null ? jsonencode(var.configuration) : null
+  configuration     = var.configuration != null ? jsonencode({ for k, v in var.configuration : k => v if v != null }) : null
```

Comment:
maybe leave a code comment above this to say why we need to do this. Perhaps add the root cause into https://github.com/terraform-ibm-modules/terraform-ibm-icd-postgresql/issues/273 and just link to that issue?